src/IssueViz/IssueVisualizationControl/ViewModels/FileNameLocationListItem.cs

```diff
@@ -68,8 +70,18 @@ private void Location_PropertyChanged(object sender, PropertyChangedEventArgs e)
         private void UpdateState()
         {
             FullPath = location.CurrentFilePath;
-            FileName = Path.GetFileName(FullPath);
-            Icon = GetImageMonikerForFile(FullPath);
+
+            if (FileExists)
```

Comment:
If it is really the case that the existence/non-existence of a file can be determined from whether `IAnalysisIssueLocationVisualization.CurrentFilePath" is null or not then the doc comment for that interface method should say so to make it's clear it's part of the interface contract.

Reply:
I checked what happens for regular issues: we don't support it.
If you delete a file for a secondary location, nothing will change in the panel, and any attempt to navigate will simply become non-navigable.
This is because we don't actually do anything when it happens: https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/IssueViz/Editor/FileRenamesEventSource.cs#L102

So for live issues this is not relevant: CurrentFilePath might not be up-to-date (i.e. if it was deleted), but it will never be null.

For both hotspots and taint, it can definitely be null if the file cannot be found in the solution -- it doesn't mean that the file doesn't exist on disk, just that it wasn't found in the solution (e.g. the project could've been unloaded).

However, I do think that having a null CurrentFilePath makes sense: if we were to support file deletions for live issues, it would be null. So I will change the doc comment to say so. Let me know if you have objections.